### ISSUE TITLE
Fails to build with cabal

### ISSUE THREAD

**Issue by josephcsible** [ndmitchell/hlint#948]
When I run `cabal install` as of ea3b250, this happens:
```
src/Hint/Extensions.hs:326:28: error:
    Ambiguous occurrence `isWholeFrac'
    It could refer to
       either `Language.Haskell.GhclibParserEx.GHC.Hs.Expr.isWholeFrac',
              imported from `Language.Haskell.GhclibParserEx.GHC.Hs.Expr' at src/Hint/Extensions.hs:227:1-50
           or `Hint.Extensions.isWholeFrac',
              defined at src/Hint/Extensions.hs:515:1
    |
326 | usedExt NumDecimals = hasS isWholeFrac
    |                            ^^^^^^^^^^^

src/Hint/Extensions.hs:390:67: error:
    Ambiguous occurrence `isFieldPunUpdate'
    It could refer to
       either `Language.Haskell.GhclibParserEx.GHC.Hs.Expr.isFieldPunUpdate',
              imported from `Language.Haskell.GhclibParserEx.GHC.Hs.Expr' at src/Hint/Extensions.hs:227:1-50
           or `Hint.Extensions.isFieldPunUpdate',
              defined at src/Hint/Extensions.hs:521:1
    |
390 | used RecordPuns = hasS isPatFieldPun ||^ hasS isFieldPun ||^ hasS isFieldPunUpdate
    |                                                                   ^^^^^^^^^^^^^^^^
```
I suspected that #943 may be the culprit, but reverting that just replaced those problems with a new problem:
```
src/GHC/Util/HsExpr.hs:237:87: error:
    * Couldn't match type `GenLocated SrcSpan (Pat (GhcPass 'Parsed))'
                     with `Pat (GhcPass 'Parsed)'
      Expected type: String -> Pat (GhcPass 'Parsed)
        Actual type: String -> LPat GhcPs
    * In the second argument of `(.)', namely `strToPat'
      In the first argument of `map', namely `(noLoc . strToPat)'
      In the `m_pats' field of a record
    |
237 |       match = noLoc $ Match {m_ext=noExtField, m_ctxt=LambdaExpr, m_pats=map (noLoc . strToPat) ss, m_grhss=grhss} :: LMatch GhcPs (LHsExpr GhcPs)
    |                                                                                       ^^^^^^^^
```
To confirm this is easily reproducible, I tried it in a fresh `haskell:8.8.3` Docker container, and it happened there too.

**Comment by shayne-fletcher** [comment]
Oh dear.  I didn't anticipate this - sorry! @ndmitchell landing https://github.com/ndmitchell/hlint/pull/946 should fix this. @josephcsible, perhaps you could merge that branch locally and confirm?

**Comment by shayne-fletcher** [comment]
Alternatively, changing
```
    build-depends:
        ghc-lib-parser-ex >= 8.10.0.4 && < 8.10.1
```
to,
```
    build-depends:
        ghc-lib-parser-ex == 8.10.0.3
```
in your hlint.cabal should also overcome the issue.

**Comment by josephcsible** [comment]
#946 indeed fixes it for me. (And so does the alternative.)

**Comment by shayne-fletcher** [comment]
Ok, perfect. Thanks for checking! I assume you are now unblocked. Let me know if anything else comes up (although I don't expect it to).

**Comment by ndmitchell** [comment]
Thanks guys for figuring this out while I was out of action. I've merged #946.
